server/client.go

```diff
@@ -1567,11 +1567,13 @@ func (c *client) processPub(trace bool, arg []byte) error {
 	default:
 		return fmt.Errorf("processPub Parse Error: '%s'", arg)
 	}
+	// If number overruns an int64, parseSize() will have returned a negative value
 	if c.pa.size < 0 {
 		return fmt.Errorf("processPub Bad or Missing Size: '%s'", arg)
 	}
 	maxPayload := atomic.LoadInt32(&c.mpay)
-	if maxPayload != jwt.NoLimit && int32(c.pa.size) > maxPayload {
+	// Use int64() to avoid int32 overrun...
+	if maxPayload != jwt.NoLimit && int64(c.pa.size) > int64(maxPayload) {
```

Comment:
Are we sure here? If int32 is negative are we sure int64 cast will remove the negative bit?

Reply:
c.pa.size is not negative (we ensure of that line 1571). The reason original test would fail is because `int32(c.pa.size)` resulted in being negative because c.pa.size is bigger than 0x7FFFFFFF. The cast to int64() will keep it positive. Again, if number in protocol is overrunning int64, parseSize() would have returned a negative value (see test), and so we would have caught it line 1570.

Reply:
ok gotcha.